### PR TITLE
Adjust copy designators

### PR DESCRIPTION
**Pull request by BrettLMiller**
Justin
As you requested in this [Issue](https://github.com/Altium-Designer-addons/scripts-libraries/issues/112).

My latest POC version V3 of AdjustDesignators is not public & is very different to V2..
So I've patched up the last V2 version using the old method circa Oct 2021. This sort-of-works okay. This is committed to a [branch in my fork](https://github.com/Altium-Designer-addons/scripts-libraries/compare/master...BrettLMiller:scripts-libraries:AdjustCopyDesignators)..
This branch also contains the latest copy of the redundant partner script CopyDesignatorsToMechLayerPair.

### CODE REVIEW

File: Scripts - PCB/AdjustDesignators2/AdjustDesignators2.pas [hunk 1]
```diff
@@ -496,21 +510,25 @@ procedure TFormAdjustDesignators.ButtonOKClick(Sender: TObject);
...
 
-            Designator.Size := Size;
             Designator.SetState_XSizeYSize;
             Designator.GraphicallyInvalidate;
             Component.Name.EndModify;
 
+//     Set the Designator AutoPosition to the center-center but stop comment being moved.
             Component.ChangeCommentAutoposition( eAutoPos_Manual );
             Component.ChangeNameAutoposition(eAutoPos_CenterCenter);
             Component.SetState_NameAutoPos(eAutoPos_CenterCenter);
```
Comment (RnDMonkey) [inline Scripts - PCB/AdjustDesignators2/AdjustDesignators2.pas:529]: Does following `ChangeNameAutoposition(eAutoPos_CenterCenter)` with `SetState_NameAutoPos(eAutoPos_CenterCenter)` do something particular? Does it avoid having to do `EndModify` immediately followed by `BeginModify` to get object property cache to update for later manipulation?

**Review by RnDMonkey — COMMENTED** [review]
Changes look reasonable to me. @justinmassiot are you going to test and merge?

File: Scripts - PCB/AdjustDesignators2/AdjustDesignators2.pas (hunk 1 above)
Comment (BrettLMiller) [reply Scripts - PCB/AdjustDesignators2/AdjustDesignators2.pas:529]: Yes there is a reason. This script uses the hacky method of reusing (abusing) the real designator. The autoposition functions of designator & comment interfere with each other so I use a property shuffle trick to stop Altium from moving anything.

Comment (RnDMonkey) [reply Scripts - PCB/AdjustDesignators2/AdjustDesignators2.pas:529]: Yeah, I understand that part, where you set the comment to manual autoposition so it doesn't move after the designator "gets out of the way", what I'm not clear on is the double-setting of the designator autoposition using both `ChangeNameAutoposition` function and the `SetState_NameAutoPos` procedure, one right after the other. It looks redundant, but I'm wondering if it actually has a useful effect.

In my scripting I've noticed that certain changes to autoposition don't really "take effect" until I call EndModify on the text, and `ChangeNameAutoposition` seems to work more immediately. I was just wondering if you had some insight or experience on the nuance behind these calls that I'm still trying to grok.

Comment (BrettLMiller) [reply Scripts - PCB/AdjustDesignators2/AdjustDesignators2.pas:529]: Yeah, I think that extra SetState_NameAutoPos is redundant. It does not trigger the action like ChangeName & AutoPosition_NameComment do.
SetState_NameAutoPos  (fortunately) allows you to set the original auto setting back without damage.
That block of code was copied from my original designator script that uses Comment as a 2nd designator.

**Comment by justinmassiot** [comment]
Thank you @BrettLMiller!
No, I didn't test it @RnDMonkey, I have very little free time these days. Anyway, thank you for the merge; if it's broken, we're gonna hotfix it :wink:

**Pull request merged**